adm_program/system/classes/filesystemutils.php

```diff
@@ -0,0 +1,1016 @@
+<?php
+/**
+ ***********************************************************************************************
+ * Class handle the most necessary file-system operations
+ *
+ * @copyright 2004-2017 The Admidio Team
+ * @see https://www.admidio.org/
+ * @license https://www.gnu.org/licenses/gpl-2.0.html GNU General Public License v2.0 only
+ ***********************************************************************************************
+ */
+
+/**
+ * @class FileSystemUtils
+ * Class handle the most necessary file-system operations
+ */
+final class FileSystemUtils
+{
+    const CONTENT_TYPE_DIRECTORY = 'directory';
+    const CONTENT_TYPE_FILE      = 'file';
+    const CONTENT_TYPE_LINK      = 'link';
+
+    const ROOT_ID = 0;
+    const ROOT_FOLDER = '/';
+
+    private static $workingDirectories = array();
+
+    /**
+     * @param array<int,string> $directoryPaths
+     * @throws \UnexpectedValueException
+     */
+    public static function setWorkingDirectories(array $directoryPaths = array())
```

Comment:
Name the method like "setValidDirectories" Sounds better for me. "Working" is understood in other context

Reply:
Maybe "Allowed" instead of "Valid"?

Reply:
Yes, also ok